src/sentry_plugins/trello/plugin.py

```diff
@@ -0,0 +1,260 @@
+            'type': 'secret',
+            'required': True,
+        }
+        token_value = get_from_initial(initial, 'token')
+
+        if token_value:
+            token['has_saved_value'] = True
+            token['prefix'] = token_value[:6]
+            token['required'] = False
+
+        config = [key, token]
+
+        if key_value and token_value:
+            trello = TrelloClient(key_value, token_value)
+            organizations = tuple()
+            try:
+                organizations = trello.organizations_to_options()
+                organization_value = self.get_option('organization', project)
+                if not organization_value:
+                    organizations = EMPTY + organizations
+                if get_from_initial(initial, 'organization') or kwargs.get('add_additial_fields'):
+                    config.append({
+                        'name': 'organization',
+                        'label': 'Trello Organization',
+                        'type': 'select',
+                        'choices': organizations,
+                        'default': organization_value,
+                        'required': True,
+                    })
+            except RequestException as exc:
+                if exc.response is not None and exc.response.status_code == 401:
+                    self.client_errors.append(self.error_messages['invalid_auth'])
+                else:
+                    self.client_errors.append(self.error_messages['api_failure'])
+        return config
+
+    def get_group_urls(self):
+        return super(TrelloPlugin, self).get_group_urls() + [
+            (
+                r'^autocomplete', IssueGroupActionEndpoint.as_view(
+                    view_method_name='view_autocomplete',
+                    plugin=self,
+                )
+            ),
+        ]
+
+    def is_configured(self, request, project, **kwargs):
+        return all((
+            self.get_option(key, project)
+            for key in ('key', 'token', 'organization')
+        ))
+
+    def has_workspace_access(self, workspace, choices):
+        for c, _ in choices:
+            if workspace == c:
+                return True
+        return False
+
+    def get_board_choices(self, boards):
+        return [(board['id'], board['name']) for board in boards]
+
+    def get_new_issue_fields(self, request, group, event, **kwargs):
+        fields = super(TrelloPlugin, self).get_new_issue_fields(request, group, event, **kwargs)
+        client = self.get_client(request.user)
+        boards = client.get_boards()
+        board_choices = self.get_board_choices(boards)
+
+        return fields + [
+            {
+                'name': 'board',
+                'label': 'Board',
+                'type': 'select',
+                'choices': board_choices,
+                'readonly': False,
+                'required': True,
+            },
+            {
+                'name': 'list',
+                'depends': ['board'],
```

Comment:
So this is the solution I came up with @lauryndbrown for dynamic forms ^ + https://github.com/getsentry/sentry/pull/9136